### ISSUE TITLE
CryptoAnalysis gets StackOverflowError when analyzing it's own jar file

### ISSUE THREAD

**Issue by Projucti** [CodeShield-Security/SPDS#29]
The problem when Analysis runs it's own jar file. StackOverflowError is found while investigating the issue.  The snapshot of the error is here- 

![image_2021_07_27T15_52_40_271Z](https://user-images.githubusercontent.com/4972596/128672399-1e7d6361-c41f-427f-8693-89de4674bac1.png)
This issue is found while investigating [main issue](https://github.com/CROSSINGTUD/CryptoAnalysis/issues/54) from CryptoAnalysis.

**Comment by johspaeth** [comment]
@Projucti 
1. CryptoAnalysis still build on Version 2 of SPDS and needs to be ported to Version 3 (of this repository). I am assuming that has been done so far?
2. Did you increase the JVMs stack size for running CryptoAnalysis (jvm arg: `-Xss100m`)?

**Comment by Projucti** [comment]
Yes. CryptoAnalysis still works on version 2 of SPDS. Now, I am trying to work on upgrading to version 3 and looks like I am facing many compilation error due to some missing package that doesn't exist anymore. Is there any guideline how to work with the new upgrades?
Also, unfortunately the problem persisted even after increasing the JVM stack size.
